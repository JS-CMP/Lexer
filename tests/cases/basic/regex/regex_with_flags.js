var emailPattern = /^[a-z]+@[a-z]+\.[a-z]+$/i;
var phonePattern = /\d{3}-\d{3}-\d{4}/;

