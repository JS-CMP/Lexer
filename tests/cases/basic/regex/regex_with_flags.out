#include "types/JsAny.hpp"
#include "global/global.hpp"
#include "types/objects/Types.hpp"
#include "customOperators/CustomOperators.hpp"

int main() {
    JS::Any emailPattern, phonePattern;
    emailPattern = JS::Any(std::make_shared<JS::RegExp>(JS::Any(u"^[a-z]+@[a-z]+\.[a-z]+$"), JS::Any(u"i")));
    phonePattern = JS::Any(std::make_shared<JS::RegExp>(JS::Any(u"\d{3}-\d{3}-\d{4}"), JS::Any(u"")));
    return 0;
}
